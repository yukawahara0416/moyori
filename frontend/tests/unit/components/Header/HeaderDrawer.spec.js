import { shallowMount } from '@vue/test-utils'
import Component from '@/components/Header/HeaderDrawer.vue'

let wrapper
let propsData

beforeEach(() => {
  propsData = {
    value: false
  }

  wrapper = shallowMount(Component, {
    propsData
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('props', () => {
  it('value', () => {
    expect(wrapper.props().value).toStrictEqual(propsData.value)
    expect(typeof wrapper.vm.$props.value).toBe('boolean')
  })
})

describe('emit', () => {
  it('value', () => {
    wrapper.vm.input()
    expect(wrapper.emitted('input')).toBeTruthy()
  })
})

describe('template', () => {
  it('snapshot', () => {
    expect(wrapper.vm.$el).toMatchSnapshot()
  })
})
