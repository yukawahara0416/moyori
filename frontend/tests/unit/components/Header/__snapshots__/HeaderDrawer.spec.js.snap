// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`template snapshot 1`] = `
<v-navigation-drawer-stub
  absolute="true"
  app="true"
  height="100vh"
  minivariantwidth="56"
  mobilebreakpoint="1264"
  right="true"
  src=""
  tag="nav"
  temporary="true"
  width="256"
>
  <v-list-stub
    tag="div"
    tile="true"
  >
    <header-drawer-pages-stub />
     
    <v-divider-stub />
     
    <header-drawer-author-stub />
     
    <header-drawer-footer-stub />
  </v-list-stub>
</v-navigation-drawer-stub>
`;
